### PR TITLE
BSM PartII Data Redaction

### PR DESCRIPTION
**Pull request by dmccoystephenson**
# Purpose
The changes in this PR allow the BSMHandler to redact specified fields in the partII section of BSM messages.

## Notes
- The BSMHandler makes use of the fieldsToRedact.txt file, which is located the project root folder. 
- Three new tests have been introduced, one to test the RedactionPropertyManager and two to test the partII redaction.
- At this time, an instance of the RedactionPropertyManager class is initialized with each BSM message that goes through processing. Each time it is initialized, the fields are loaded in. We may want to consider making this a part of the BSMHandler class if we can to minimize the number of times that the fields are loaded in.

### CODE REVIEW

File: src/tests.cpp [hunk 1]
```diff
@@ -1424,3 +1438,52 @@ TEST_CASE( "BSMHandler JSON Error Checking", "[ppm][filtering][error]" ) {
...
+        // process test case to build BSM (redaction will occur here)
+        CHECK( handler.process( test_case ) );
+
+        // make sure that it was successful
+        CHECK( handler.get_result_string() == "success" );
+
+        // verify that there are no sensitive members in the partII field left
+        std::string partIIString = handler.get_bsm().get_partII();
+        rapidjson::Document partII;
+        partII.Parse(partIIString.c_str());
```
Comment (dmccoystephenson) [inline src/tests.cpp:1479]: This needs to be verified to have correctly parsed the partII string. If it is not getting parsed correctly, which I suspect is the case, then it's likely that we need to alter the partIIString before we parse it.

Comment (dmccoystephenson) [reply src/tests.cpp:1479]: To clarify, redaction functionality has been verified to be working via debug messages, but I don't think the above code will cause the test to fail (as it should) when there are still fields that should have been redacted remaining in the partII section of the BSM message.

Comment (dmccoystephenson) [reply src/tests.cpp]: This test will now fail upon failure to parse and upon finding sensitive fields in the partII section of the BSM message after redaction. The test also requires that the partIIString not be empty.

File: src/redaction-properties/RedactionPropertiesManager.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            }
+            
+            log("redaction fields loaded");
+        }
+
+        /**
+         * @brief Saves the fields to a file.
+         * 
+         */
+        void saveFields(std::string fileName) {
```
Comment (dmccoystephenson) [inline src/redaction-properties/RedactionPropertiesManager.cpp:169]: Will we want to be able to save fields added during runtime?

**Comment by dmccoystephenson** [comment]
At this time, we only have one test BSM that is getting tested for partII redaction. We might want to add some more.

File: src/redaction-properties/RedactionPropertiesManager.cpp (hunk 2 above)
Comment (dmccoystephenson) [reply src/redaction-properties/RedactionPropertiesManager.cpp]: I've removed the skeleton methods in this class for now.

**Review by drewjj — APPROVED** [review]

**Comment by drewjj** [comment]
Looks good!

**Pull request merged**